### PR TITLE
Adopting Bootstrap 4 as option

### PR DESCRIPTION
**Pull request by MarkyMarkNo1**
Hi together,

please accept my pull request.

I added an additional choice for Bootstrap v3 or v4.

Bootstrap 3 is treated as 'legacyBootstrap'. Everything Bootstrap 3 related is wrapped in applicable conditions, that can be safely removed once we decide to drop Boostrap 3.

Thank you!

### CODE REVIEW

File: app/templates/index.html [hunk 1]
```diff
@@ -23,22 +23,28 @@
...
+          <li>
+            <a <% if (!legacyBootstrap) { %>class="nav-link" <% } %>href="#">Contact</a>
+          </li>
         </ul>
         <h3 class="text-muted"><%= appname %></h3>
       </div>
 
       <div class="jumbotron">
-        <h1>'Allo, 'Allo!</h1>
+        <h1 <% if (!legacyBootstrap) { %>class="display-3" <% } %>>'Allo, 'Allo!</h1>
```
Comment (silvenon) [inline app/templates/index.html]: Move the space after `h1` inside the `if` statement:

```html
<h1<% if (!legacyBootstrap) { %> class="display-3" <% } %>>'Allo, 'Allo!</h1>
```

(Notice the space before `class`.)

File: app/templates/gulpfile.js [hunk 2]
```diff
@@ -64,9 +64,18 @@ gulp.task('html', ['styles'], () => {
...
+    .pipe($.if('*.html', $.htmlmin({
+      collapseWhitespace: true,
+      minifyCSS: true,
+      minifyJS: {compress: {drop_console: true}},
+      processConditionalComments: true,
+      removeComments: true,
+      removeEmptyAttributes: true,
+      removeScriptTypeAttributes: true,
+      removeStyleLinkTypeAttributes: true
+    })))
```
Comment (silvenon) [inline app/templates/gulpfile.js:78]: Could you explain these changes to the `html` task?

**Review by silvenon — APPROVED** [review]
I like this, and it works.

I was going to wait until Bootstrap got out of alpha, but it doesn't appear to be coming soon and v3 is becoming kind of deprecated.

I left two comments, after resolving them I will merge.

**Comment by silvenon** [comment]
Also, please fix the tests if you can.

**Comment by MarkyMarkNo1** [comment]
Thank you for your quick reply! I'm excited 😉

Yes, [with Alpha6 I think it's the time to get engaged](http://blog.getbootstrap.com/2017/01/06/bootstrap-4-alpha-6/#up-next-our-first-beta) with Bootstrap 4:

> As we head to our first beta, we’ll be focused on not adding anything new, ideally making as few breaking changes as possible

Regarding your remarks:

> Notice the space before class

I've been recognizing this after I created the pull request - compliment for your sharp eye!

> Could you explain these changes to the html task?

Sure, my suggestions. Would you like inline comments, or just here for clarification?

```javascript
// https://github.com/kangax/html-minifier#options-quick-reference
$.htmlmin({
  collapseWhitespace: true,
  // Minify CSS in style elements and style attributes
  minifyCSS: true,
  // Minify JavaScript in script elements and event attributes
  minifyJS: {compress: {drop_console: true}},
  // Process contents of conditional comments through minifier
  processConditionalComments: true,
  // Strip HTML comments
  removeComments: true,
  // Remove all attributes with whitespace-only values
  removeEmptyAttributes: true,
  // Remove type="text/javascript" from script tags
  removeScriptTypeAttributes: true,
  // Remove type="text/css" from style and link tags
  removeStyleLinkTypeAttributes: true
})
```

> Also, please fix the tests if you can

I'll try ...

File: app/templates/gulpfile.js (hunk 2 above)
Comment (MarkyMarkNo1) [reply app/templates/gulpfile.js:78]: Sure, my suggestions. Would you like inline comments, or just here for clarification?

```javascript
// https://github.com/kangax/html-minifier#options-quick-reference
$.htmlmin({
  collapseWhitespace: true,
  // Minify CSS in style elements and style attributes
  minifyCSS: true,
  // Minify JavaScript in script elements and event attributes
  minifyJS: {compress: {drop_console: true}},
  // Process contents of conditional comments through minifier
  processConditionalComments: true,
  // Strip HTML comments
  removeComments: true,
  // Remove all attributes with whitespace-only values
  removeEmptyAttributes: true,
  // Remove type="text/javascript" from script tags
  removeScriptTypeAttributes: true,
  // Remove type="text/css" from style and link tags
  removeStyleLinkTypeAttributes: true
})
```

**Comment by MarkyMarkNo1** [comment]
@silvenon Tests added, checks passed. Thank you!

File: app/templates/gulpfile.js (hunk 2 above)
Comment (silvenon) [reply app/templates/gulpfile.js:78]: I meant, how are these changes related to Bootstrap 4?

**Comment by silvenon** [comment]
@MarkyMarkNo1 awesome! Lastly, I just wanted to know how the changes to the `html` task are related to Bootstrap 4.

File: app/templates/gulpfile.js (hunk 2 above)
Comment (MarkyMarkNo1) [reply app/templates/gulpfile.js:78]: Actually these tweaks are not in direct relation to BS4, just best practices to optimize a production build.

**Comment by silvenon** [comment]
Thanks! 👍

**Pull request merged**